security-oauth2/src/main/java/io/micronaut/security/oauth2/endpoint/DefaultSecureEndpoint.java

```diff
@@ -29,14 +31,14 @@
 public class DefaultSecureEndpoint implements SecureEndpoint {
 
     private final String url;
-    private final List<AuthenticationMethod> supportedAuthenticationMethods;
+    private final List<String> supportedAuthenticationMethods;
 
     /**
      * @param url The endpoint URL
      * @param supportedAuthenticationMethods The endpoint authentication methods
      */
     public DefaultSecureEndpoint(@NonNull String url,
```

Comment:
Should we keep the previous constructor or is it internal class?

Reply:
We cannot. Java does not allow both constructors. 

```java
public DefaultSecureEndpoint(@NonNull String url,
                                 @Nullable List<AuthenticationMethod> supportedAuthenticationMethods) {
public DefaultSecureEndpoint(@NonNull String url,
                                 @Nullable List<String> supportedAuthenticationMethods) {
```
We get a compilation method complaining about both methods have same the erasure.

Reply:
Since, I changed to` Set<String>` I was able to keep the old constructor.